documentation/modules/ROOT/pages/connectors/mongodb.adoc

```diff
@@ -1022,6 +1022,19 @@ You can use the {prodname} connector for MongoDB with https://www.mongodb.com/at
 Note that MongoDB Atlas only supports secure connections via SSL, i.e. the xref:mongodb-property-mongodb-ssl-enabled[`+mongodb.ssl.enabled`] connector option _must_ be set to `true`.
 endif::community[]
 
+
+[[mongodb-optimal-oplog-config]]
+=== Optimal Oplog Config
+
+
+The {prodname} MongoDB connector works by obtaining changes from https://www.mongodb.com/docs/manual/core/replica-set-oplog/[Oplog] using https://www.mongodb.com/docs/manual/changeStreams/[change streams]. As oplog is a https://www.mongodb.com/docs/manual/reference/glossary/#std-term-capped-collection/[capped collection], older entries are overwritten once maximum configured size is reached. For MongoDB connector to be able to deal with failures it is important that the oplog retention config is such that the resume point is available upon restart or retries. If the last oplog position is lost, the connector can not start again and a new connector has to be created afresh - See xref:debezium-mongodb-connector-is-stopped-for-a-long-interval[Connector fails after it is stopped for a long interval if snapshot.mode is set to initial].
+
+To avoid this, you can provide a higher https://www.mongodb.com/docs/manual/core/replica-set-oplog/[Oplog size] or preferably a https://www.mongodb.com/docs/manual/core/replica-set-oplog/#std-label-replica-set-minimum-oplog-size/[Minimum Oplog Window] config available since MongoDB 4.4 (i.e. time-based, entries of the last X-hours are guaranteed to be available even if oplog size cap is reached). The latter is preferable but may not always be an option as it can have impact if the cluster is reaching its capacity and there is sudden load of data. Thus, you can instead choose to provide a higher Oplog size based on you workloads, ideally should be more than peak Oplog/Hour.
+
+E.g. If we have 3 GB oplog per hour and only 1 GB oplog size limit. Oplog entries will get cleared thrice an hour, which would be bad if Debezium fails during this time. Thus, if your average reaction time to Debezium failure is 2 hours then the oplog size should be at least 6 GB.
+
+Monitoring Oplog is easy on a Atlas MongoDB Cluster, See https://www.mongodb.com/basics/how-to-monitor-mongodb-and-what-metrics-to-monitor#oplog-gbhour/[Monitoring Oplog] about Replication Oplog Window (if debezium is disabled and lagging more than this, then it is bound to fail) and Oplog GB/Hour (explained above).
```

Comment:
I think Replication Oplog Window is an important metric to look for in case of failure. I'll check how we can improve the above to add that - probably we should add that in the beginning itself. Any specific reason for removing it other than simplifying the text?

Reply:
Yes, my intention was to simplify the text. The original sentence was a bit to dense, and its meaning was not entirely clear. The original link text refers generally to _Monitoring Oplog_ , but the link target is actually _Oplog GB/Hour_, so it seemed like the important takeaway for readers was that they should monitor replication metrics. I added a link in my [previous edit](https://github.com/debezium/debezium/pull/4598#discussion_r1235492479) to the Oplog GB/Hour metric, so readers who follow that link would land in the same _MongoDB Replication Metrics_ section that discusses the Replication Oplog Window.

I've added a couple of sentences to my original edit to suggest the specific metrics to track.